webapp/src/Entity/AuditLog.php

```diff
@@ -15,71 +15,63 @@
 class AuditLog
 {
     /**
-     * @var int
-     *
      * @ORM\Id
      * @ORM\GeneratedValue(strategy="AUTO")
      * @ORM\Column(type="integer", name="logid", length=4,
      *     options={"comment"="Audit log ID","unsigned"=true}, nullable=false)
      */
-    private $logid;
+    private ?int $logid;
 
     /**
-     * @var double
+     * @var double|string
      * @ORM\Column(type="decimal", precision=32, scale=9, name="logtime", options={"comment"="Timestamp of the logentry", "unsigned"=true}, nullable=false)
      */
     private $logtime;
 
     /**
-     * @var int
      * @ORM\Column(type="integer", name="cid", length=4,
      *     options={"comment"="Contest ID associated to this entry",
      *              "unsigned"=true},
      *     nullable=true)
      */
-    private $cid;
+    private ?int $cid;
 
     /**
-     * @var string
      * @ORM\Column(type="string", name="user", length=255,
      *     options={"comment"="User who performed this action"},
      *     nullable=true)
      */
-    private $user;
+    private ?string $user;
 
     /**
-     * @var string
      * @ORM\Column(type="string", name="datatype", length=32,
      *     options={"comment"="Reference to DB table associated to this entry"},
      *     nullable=true)
      */
-    private $datatype;
+    private string $datatype;
```

Comment:
nullable is true, does that imply $datatype = null?

Reply:
Yes. Apparently it is never null so maybe at some point we should make it nonnullable though